### PR TITLE
refactor: organize filter sort query into folders in client

### PR DESCRIPTION
**Pull request by nathansandi**
## Description

Separate the search request and impl into folders
- Sort
- Filter
- Query (Domain files plus Query setup requirements)

Setup/Builder ones was keep into the root.

## Checklist

- [x] for CI changes:
  - [x] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [x] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

### CODE REVIEW

File: clients/java/src/main/java/io/camunda/zeebe/client/api/search/query/SearchRequestBuilders.java [hunk 1]
```diff
@@ -13,18 +13,27 @@
...
-import io.camunda.zeebe.client.impl.search.DecisionDefinitionSortImpl;
-import io.camunda.zeebe.client.impl.search.DecisionRequirementsFilterImpl;
-import io.camunda.zeebe.client.impl.search.DecisionRequirementsSortImpl;
-import io.camunda.zeebe.client.impl.search.ProcessInstanceFilterImpl;
-import io.camunda.zeebe.client.impl.search.ProcessInstanceSortImpl;
-import io.camunda.zeebe.client.impl.search.SearchRequestPageImpl;
-import io.camunda.zeebe.client.impl.search.UserTaskFilterImpl;
-import io.camunda.zeebe.client.impl.search.UserTaskSortImpl;
-import io.camunda.zeebe.client.impl.search.VariableValueFilterImpl;
+package io.camunda.zeebe.client.api.search.query;
```
Comment (nicpuppa) [inline clients/java/src/main/java/io/camunda/zeebe/client/api/search/query/SearchRequestBuilders.java:16]: 🔧 This file shouldn't be in this package IMO. I would move it into the root of `/impl/search`

File: clients/java/src/main/java/io/camunda/zeebe/client/api/search/query/SearchRequestPage.java [hunk 2]
```diff
@@ -13,7 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.camunda.zeebe.client.api.search;
+package io.camunda.zeebe.client.api.search.query;
```
Comment (nicpuppa) [inline clients/java/src/main/java/io/camunda/zeebe/client/api/search/query/SearchRequestPage.java:16]: 🔧 This file shouldn't be in this package. I would move it outside the query package and leave it into the `/search` root

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/search/query/SearchRequestPageImpl.java [hunk 3]
```diff
@@ -13,9 +13,9 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.camunda.zeebe.client.impl.search;
+package io.camunda.zeebe.client.impl.search.query;
```
Comment (nicpuppa) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/search/query/SearchRequestPageImpl.java:16]: 🔧 I'll move this class in the `/impl/search` root

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/search/query/SearchResponseMapper.java [hunk 4]
```diff
@@ -13,7 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.camunda.zeebe.client.impl.search;
+package io.camunda.zeebe.client.impl.search.query;
```
Comment (nicpuppa) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/search/query/SearchResponseMapper.java:16]: 🔧 I'll move this class in the `/impl/search` root

File: clients/java/src/main/java/io/camunda/zeebe/client/impl/search/query/TypedSearchRequestPropertyProvider.java [hunk 5]
```diff
@@ -13,7 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.camunda.zeebe.client.impl.search;
+package io.camunda.zeebe.client.impl.search.query;
```
Comment (nicpuppa) [inline clients/java/src/main/java/io/camunda/zeebe/client/impl/search/query/TypedSearchRequestPropertyProvider.java:16]: 🔧 I'll move this class in the `/impl/search` root

**Review by nicpuppa — COMMENTED** [review]
Hey @nathansandi great refactoring 🚚 

I left some comments please have a look 👀 🍪

**Comment by nathansandi** [comment]
> Hey @nathansandi great refactoring 🚚
> 
> I left some comments please have a look 👀 🍪

Thank you for the suggestions @nicpuppa  :) I applied them.

**Review by nicpuppa — APPROVED** [review]
Thanks @nathansandi 💪

**Pull request merged**